### ISSUE TITLE
FPM tags

### ISSUE THREAD

**Issue by frankvanhest** [WyriHaximusNet/docker-php#210]
Hi @WyriHaximus,

In the documentation fpm is mentioned, but in the tags I can't seem to find a tag for fpm on docker hub.
Hopefully, you can help me further.

Cheers,

Frank

**Comment by WyriHaximus** [comment]
Hey @frankvanhest I'll update the docs to remove that, this repo isn't building FPM images. But it is probably a leftover from when I based this repo on https://github.com/usabilla/php-docker-template

**Comment by frankvanhest** [comment]
Thanks for your quick response! I'll try to install it manually in the container. I love the builds you provide, fpm was the only thing missing.

**Comment by WyriHaximus** [comment]
The main goal of these images are for running ReactPHP from the CLI, what are you looking for in them for FPM?

**Comment by frankvanhest** [comment]
I know they are build for ReactPHP and that is something you don't really use in FPM. I don't intend to use it in FPM. Your images are complete with all the most common PHP extensions, composer included, Xdebug included, and so on. I was hoping that FPM was included, so I can use one Docker hub for everything PHP related in my projects. Thats the main reason for me

**Comment by WyriHaximus** [comment]
Let's get started that I'm not opposed to adding FPM images. Might have rejected it in the past but IMHO it could serve for micro ReactPHP apps during FPM requests as well.

Would you be interested in picking this task up? There is some work that would needs to be done from Docker files to tests to dealing with odd exceptions.

**Comment by frankvanhest** [comment]
Yeah sure, I can give it a try! It can give me a better understanding on how to build docker images and ReactPHP.
I can get started on it this Friday night.

**Comment by WyriHaximus** [comment]
Awesome! Just ping me when you have questions or run into any thing

**Comment by frankvanhest** [comment]
@WyriHaximus, well I'm a bit stuck with adding FPM. I created the Docker files for Alpine an Debian. Running the makefile gives some error due to missing environment variables such as GITHUB_OUTPUT used in https://github.com/WyriHaximus/github-action-supported-php-versions/blob/master/versions.php

Do you have the time to help get underway?

**Comment by WyriHaximus** [comment]
@frankvanhest Haven't ran into that one before. Will have a look tomorrow morning at you branch and see if I have the same issue

**Comment by WyriHaximus** [comment]
@frankvanhest Looks like `/` isn't the default path you enter the container in for FPM. So change line 25 to:

```dockerfile
RUN git clone https://github.com/amphp/ext-uv uv
```

**Comment by frankvanhest** [comment]
@WyriHaximus Thanks for your help! The work directory was indeed incorrect. I The line you mentioned was already correct. After I changed line 26 to 
```
WORKDIR uv
```
I could successfully build the image. 

<img width="649" alt="Screenshot 2022-11-26 at 14 19 03" src="https://user-images.githubusercontent.com/2206481/204091001-a1c11115-20a6-43d6-aa35-3d944834f231.png">

Is there a way to run the `make build` command locally? It looks like depends on GitHub actions and their environment. 
Where does the file `docker-image/image.tags` gets created? By looking a the code, I cannot seem to find it.

**Comment by WyriHaximus** [comment]
So everything is currently in a state of transition to a better solution for both local and GHA building. And while most of it is working on GHA, the same scripts can be used for the Makefile, but it needs some work. Will have a look at it after dinner.

Planning a few steps a head here, but can you file a PR like this one https://github.com/WyriHaximusNet/docker-php/pull/209 but for RC7? This way once you file the PR for the FPM tags you don't need to wait for my approval to run the workflow :)

**Comment by frankvanhest** [comment]
Thanks! I've created one for RC7 https://github.com/WyriHaximusNet/docker-php/pull/211. Tomorrow evening I'll continue working on this.

**Comment by WyriHaximus** [comment]
I think you PR'ed the wrong changes ;). They introduced that when cryptocurrency fanatics where filing PR's to run miners in GHA 🤐 . And it marks you as cool after the first merged PR 😅
